src/types.ts

```diff
@@ -46,6 +46,7 @@ export interface ProgrammableContractFunction extends WatchableContractFunction
   revertsAtCall: (index: number, reason?: string) => void;
   whenCalledWith: (...args: unknown[]) => WhenCalledWithChain;
   reset: () => void;
+  (...args: any[]): Promise<any>;
```

Comment:
Why would you want this? In any case, it should be any, it might not return a promise
```suggestion
  (...args: any[]): any;
```